test/functional/Specification/SpecificationLoaderTest.php

```diff
@@ -0,0 +1,356 @@
+                    }
+
+                    /**
+                     * {@inheritDoc}
+                     */
+                    public function tag($tags): ItemInterface
+                    {
+                        return $this;
+                    }
+
+                    /**
+                     * {@inheritDoc}
+                     */
+                    public function expiresAt($expiration)
+                    {
+                        return $this;
+                    }
+                });
+                $this->items[$key] = $item;
+
+                return $item;
+            }
+
+            /**
+             * @return array|Specification[]
+             */
+            public function getCachedItems(): array
+            {
+                return $this->items;
+            }
+
+            public function set(string $key, Specification $item): void
+            {
+                $this->items[$key] = $item;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            public function invalidateTags(array $tags)
+            {
+                return true;
+            }
+
+            public function delete(string $key): bool
+            {
+                return true;
+            }
+        };
+
+        parent::setUp();
+    }
+
+    public function tearDown(): void
+    {
+        unset($this->specificationParser, $this->fileLocator, $this->cache);
+    }
+
+    public function testRegisterSpecAndGettingRegisteredSpecifications(): void
+    {
+        $specificationLoader         = new SpecificationLoader(
+            new SpecificationParser(new ScalarTypesResolver()),
+            $this->fileLocator,
+            $this->cache
+        );
+        $specificationFileName       = 'specification.json';
+        $specificationArray          = $this->getSpecificationArray($specificationFileName);
+        $expectedSpecificationConfig = new SpecificationConfig(
+            $specificationArray['path'],
+            $specificationArray['type'] ?? null,
+            $specificationArray['name_space'],
+            $specificationArray['media_type']
+        );
+
+        $expectedSpecifications = [self::SPECIFICATION_NAME => $expectedSpecificationConfig];
+
+        $specificationLoader->registerSpec(self::SPECIFICATION_NAME, $specificationArray);
+        $specifications = $specificationLoader->list();
+
+        Assert::assertEquals($expectedSpecifications, $specifications);
+    }
+
+    public function testGetThrowsException(): void
+    {
+        $specificationName   = 'test';
+        $specificationLoader = new SpecificationLoader(
+            $this->specificationParser,
+            $this->fileLocator,
+            $this->cache
+        );
+
+        $this->expectException(Throwable::class);
+        $this->expectExceptionMessage(sprintf('OpenApi spec "' . $specificationName . '" is not registered in bundle config, ' .
+            'Registered specs are: .'));
```

Comment:
The list of registered specs in the exception message should also be checked.
Register a spec in the loader and test with a spec name that was not registered.